blockchain/chainquery.go

```diff
@@ -48,13 +48,11 @@ func (b *BlockChain) ChainTips() []dcrjson.GetChainTipsResult {
 	}
 	b.index.RUnlock()
 
-	b.chainLock.Lock()
-	bestTip := b.bestNode
-	b.chainLock.Unlock()
-
 	// Generate the results sorted by descending height.
 	sort.Sort(sort.Reverse(nodeHeightSorter(chainTips)))
 	results := make([]dcrjson.GetChainTipsResult, len(chainTips))
+	b.chainLock.RLock()
```

Comment:
what is being protected by this mutex (why is being held longer here now?)

Reply:
The `inMainChain` flag on the node is protected by the chain lock.

Reply:
Nevermind, that's a different function I'm thinking about. This one is protecting `b.bestNode`.  However, it could certainly be limited to just that field.  I'll take care of that.

Reply:
Done.